From: Drew Parsons <dparsons@debian.org>
Date: Fri, 9 May 2008 13:20:11 +0000 (+1000)
Subject: Disable D-BUS from Xprint.
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/xserver.git;a=commitdiff;h=2a3d1421e0cc18822ae8f478fcc272e16a9e9340

Disable D-BUS from Xprint.

Use dummy config functions to replace those from config/config.c, and
therefore do not link Xprt with $CONFIG_LIB.

Works around an endlessly spinning loop in dix/dispatch.c::Dispatch()
(WaitForSomething() not waiting) when built with dbus, which was
causing Xprt to use 95% cpu.
---

--- a/hw/xprint/ddxInit.c
+++ b/hw/xprint/ddxInit.c
@@ -291,6 +291,12 @@ ddxProcessArgument (
 #include "XIstubs.h"
 #include "exglobals.h"
 
+/* Place dummy config functions here instead of config/config.c, 
+   since Xprint does not use D-BUS */
+void config_init() { }
+void config_fini() { }
+
+
 int
 ChangePointerDevice (
     DeviceIntPtr       old_dev,
